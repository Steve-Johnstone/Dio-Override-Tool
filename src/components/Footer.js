import React from 'react';
import PropTypes from 'prop-types';
import { ImCross } from 'react-icons/im';
import { IoMdRefresh } from 'react-icons/io';
import { AiOutlineLink } from 'react-icons/ai';
import { buildUrl } from '../helpers/buildUrl';
import 'bulma/css/bulma.min.css';

const Footer = ({ url, selectedOverrides, setSelectedOverrides }) => {
	const handleClick = (clickedOverride) => {
		setSelectedOverrides(
			selectedOverrides.filter((override) => override !== clickedOverride)
		);
	};

	return (
		<div className='footer-section'>
			<div className='footer-override-display'>
				<ul className='footer-override-list'>
					{selectedOverrides.map((override) => {
						return (
							<li
								key={selectedOverrides.indexOf(override)}
								onClick={() => handleClick(override)}
								data-testid='list-item'
							>
								{override}
							</li>
						);
					})}
				</ul>
			</div>
			<div className='footer-url-display ml-5 level-item is-justify-content-flex-start'>
				<div className='ml-3'>
					<AiOutlineLink />
				</div>
				<div id='url-display' data-testid='url-display' className='p-2'>
					{buildUrl(url, selectedOverrides)}
				</div>
			</div>
			<div className='level-item is-justify-content-space-between ml-5'>
				<div>
					Click anywhere on the box to select the whole URL, this way you can
					copy it.
				</div>
				<div className='mr-5'>
					<button
						name='clear-all'
						className='button is-white m-2'
						onClick={() => setSelectedOverrides([])}
					>
						<ImCross />
						&nbsp;&nbsp;Clear All
					</button>
					<a
						id='go-button'
						className='button m-2'
						href={buildUrl(url, selectedOverrides)}
						target='blank'
					>
						<IoMdRefresh />
						&nbsp;&nbsp;Show it
					</a>
				</div>
			</div>
		</div>
	);
};

Footer.propTypes = {
	url: PropTypes.string,
	selectedOverrides: PropTypes.array,
	setSelectedOverrides: PropTypes.func,
};

export default Footer;
