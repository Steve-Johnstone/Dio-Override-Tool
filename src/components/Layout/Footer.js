import { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../UI/Button';
import SelectedOverrideList from '../UI/SelectedOverrideList';
import UrlDisplay from '../UI/UrlDisplay';
import { buildUrl } from '../../helpers/buildUrl';
import { MdClear } from 'react-icons/md';
import { IoMdRefresh } from 'react-icons/io';
import 'bulma/css/bulma.min.css';

const Footer = ({ url, selectedOverrides, setSelectedOverrides }) => {
	const [urlCopied, setUrlCopied] = useState(false);

	const handleOverrideClick = (clickedOverride) => {
		setSelectedOverrides(
			selectedOverrides.filter((override) => override !== clickedOverride)
		);
	};

	const handleCopyButtonClick = () => {
		const displayBox = 'url-display-box';
		let r = document.createRange();
		r.selectNode(document.getElementById(displayBox));
		window.getSelection().removeAllRanges();
		window.getSelection().addRange(r);
		document.execCommand('copy');
		window.getSelection().removeAllRanges();
		setUrlCopied(true);
		document.getElementById(displayBox).style.backgroundColor = '#fefee6';
		document.getElementById('copy-button').className += ' is-light';
	};

	const handleClearButtonClick = () => {
		setSelectedOverrides([]);
		setUrlCopied(false);
		document.getElementById('url-display-box').style.backgroundColor = '';
		document.getElementById('copy-button').className = 'button is-info';
	};

	return (
		<div className='footer-section'>
			<div className='footer-container'>
				<SelectedOverrideList
					selectedOverrides={selectedOverrides}
					onClick={handleOverrideClick}
				/>

				<p className='footer-label'>URL:</p>
				<UrlDisplay url={url} selectedOverrides={selectedOverrides} />

				<div className={urlCopied ? 'url-copied' : ''}>
					<div>{urlCopied ? 'URL copied to clipboard' : ''} </div>
				</div>
				<div className='level-item is-justify-content-space-between'>
					<Button
						className='is-info'
						text='Copy URL'
						id='copy-button'
						onClick={handleCopyButtonClick}
					></Button>

					<div className='level-item is-justify-content-space-between mr-5'>
						<Button
							className='is-danger m-2'
							name='clear-all'
							text='Clear All'
							onClick={handleClearButtonClick}
							icon={<MdClear />}
						></Button>

						<a
							name='show-it'
							className='button is-success m-2'
							href={buildUrl(url, selectedOverrides)}
							target='blank'
						>
							<IoMdRefresh />
							&nbsp;&nbsp;Show it
						</a>
					</div>
				</div>
			</div>
		</div>
	);
};

Footer.propTypes = {
	url: PropTypes.string,
	selectedOverrides: PropTypes.array,
	setSelectedOverrides: PropTypes.func,
};

export default Footer;
