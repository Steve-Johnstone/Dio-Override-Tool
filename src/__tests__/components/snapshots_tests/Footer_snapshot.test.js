import renderer from 'react-test-renderer';
import Footer from '../../../components/Footer';

describe('Footer component', () => {
	const setSelectedOverrides = jest.fn();

	it('should display the DOM correctly', () => {
		const tree = renderer
			.create(
				<Footer
					url={'cop/bookingdetails/bookingdetailspage'}
					selectedOverrides={['Already Charged Box', 'No Amenities']}
					setSelectedOverrides={setSelectedOverrides}
				/>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
