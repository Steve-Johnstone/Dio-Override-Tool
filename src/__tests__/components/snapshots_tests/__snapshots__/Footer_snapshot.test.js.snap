// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should display the DOM correctly 1`] = `
<div
  className="footer-section"
>
  <div
    className="footer-override-display"
  >
    <ul
      className="footer-override-list"
    >
      <li
        data-testid="list-item"
        onClick={[Function]}
      >
        Already Charged Box
      </li>
      <li
        data-testid="list-item"
        onClick={[Function]}
      >
        No Amenities
      </li>
    </ul>
  </div>
  <div
    className="footer-url-display ml-5 level-item is-justify-content-flex-start"
    id="url-display-box"
    onClick={[Function]}
  >
    <div
      className="ml-3"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
        />
      </svg>
    </div>
    <div
      className="p-2"
      data-testid="url-display"
      id="url-display"
    >
      http://localhost:8080/cop/bookingDetails/bookingDetailsPage?override=Already Charged Box,No Amenities
    </div>
  </div>
  <div
    className=""
  >
    <div>
      
       
    </div>
  </div>
  <div
    className="level-item is-justify-content-space-between ml-5"
  >
    <div>
      Click anywhere on the box above to copy the URL to the clipboard.
    </div>
    <div
      className="mr-5"
    >
      <button
        className="button is-white m-2"
        name="clear-all"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"
          />
        </svg>
          Clear All
      </button>
      <a
        className="button m-2"
        href="http://localhost:8080/cop/bookingDetails/bookingDetailsPage?override=Already Charged Box,No Amenities"
        id="go-button"
        name="show-it"
        target="blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 388c-72.597 0-132-59.405-132-132 0-72.601 59.403-132 132-132 36.3 0 69.299 15.4 92.406 39.601L278 234h154V80l-51.698 51.702C348.406 99.798 304.406 80 256 80c-96.797 0-176 79.203-176 176s78.094 176 176 176c81.045 0 148.287-54.134 169.401-128H378.85c-18.745 49.561-67.138 84-122.85 84z"
          />
        </svg>
          Show it
      </a>
    </div>
  </div>
</div>
`;
